plugins/inputs/snmp/snmp.go

```diff
@@ -363,9 +391,39 @@ func (s *Snmp) Gather(acc telegraf.Accumulator) error {
 				Name:   s.Name,
 				Fields: s.Fields,
 			}
+
 			topTags := map[string]string{}
 			if err := s.gatherTable(acc, gs, t, topTags, false); err != nil {
-				acc.AddError(fmt.Errorf("agent %s: %w", agent, err))
+				if s.Probe {
+					rt := RTable{
```

Comment:
Again, probe is required for "period" & "@timestamp" in snmp poll case.

Reply:
One issue I see this, here we want to add period and @timestamp always. Hence, a flag like probe is not required.

You think, it will be good to have a knob to control the addition of these fields?
If so, we can keep it. However, instead of 'probe', a better name is required. Probably, 'include_time_info'?

Reply:
Actually to keep it similar with heartbeat, I kept probe here as well. We can change the name of the flag easily, I think that would be good.

Reply:
Ok, let us make the name change.

Once you take care of this comment and the other points for which code changes required, please update the PR and let me know.